<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1221</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>3D Viewer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="GLWidget" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>0</y>
      <width>700</width>
      <height>700</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="file_button">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>300</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Open file</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>30</y>
      <width>301</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Rotate&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>25</y>
      <width>300</width>
      <height>3</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(0, 0, 0);</string>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>25</x>
      <y>60</y>
      <width>70</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>X (degree)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>25</x>
      <y>100</y>
      <width>70</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Y (degree)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>25</x>
      <y>140</y>
      <width>70</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Z (degree)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>190</y>
      <width>301</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Move&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>25</x>
      <y>220</y>
      <width>70</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;X&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>25</x>
      <y>300</y>
      <width>70</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Z&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>25</x>
      <y>260</y>
      <width>70</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Y&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>360</y>
      <width>301</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Scale&lt;/p&gt;&lt;p align=&quot;center&quot;&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>180</y>
      <width>300</width>
      <height>3</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(0, 0, 0);</string>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>340</y>
      <width>300</width>
      <height>3</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(0, 0, 0);</string>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>450</y>
      <width>300</width>
      <height>3</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(0, 0, 0);</string>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="OK">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>460</y>
      <width>300</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>OK</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="info">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>550</y>
      <width>300</width>
      <height>151</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="CLEAN">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>490</y>
      <width>300</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>CLEAN</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="rotate_x">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>60</y>
      <width>141</width>
      <height>25</height>
     </rect>
    </property>
    <property name="decimals">
     <number>1</number>
    </property>
    <property name="minimum">
     <double>-1000.000000000000000</double>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="rotate_y">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>100</y>
      <width>141</width>
      <height>25</height>
     </rect>
    </property>
    <property name="decimals">
     <number>1</number>
    </property>
    <property name="minimum">
     <double>-1000.000000000000000</double>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="rotate_z">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>140</y>
      <width>141</width>
      <height>25</height>
     </rect>
    </property>
    <property name="decimals">
     <number>1</number>
    </property>
    <property name="minimum">
     <double>-1000.000000000000000</double>
    </property>
    <property name="maximum">
     <double>1000.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.100000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="scale">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>390</y>
      <width>161</width>
      <height>25</height>
     </rect>
    </property>
    <property name="decimals">
     <number>2</number>
    </property>
    <property name="minimum">
     <double>0.000000000000000</double>
    </property>
    <property name="maximum">
     <double>5.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
    <property name="stepType">
     <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
    </property>
    <property name="value">
     <double>1.000000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="move_z">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>300</y>
      <width>141</width>
      <height>25</height>
     </rect>
    </property>
    <property name="decimals">
     <number>2</number>
    </property>
    <property name="minimum">
     <double>-10.000000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="move_y">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>260</y>
      <width>141</width>
      <height>25</height>
     </rect>
    </property>
    <property name="decimals">
     <number>2</number>
    </property>
    <property name="minimum">
     <double>-10.000000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="move_x">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>220</y>
      <width>141</width>
      <height>25</height>
     </rect>
    </property>
    <property name="decimals">
     <number>2</number>
    </property>
    <property name="minimum">
     <double>-10.000000000000000</double>
    </property>
    <property name="maximum">
     <double>10.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.010000000000000</double>
    </property>
   </widget>
   <widget class="QPushButton" name="saveImage">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>50</y>
      <width>221</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Save Image</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="jpeg">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>10</y>
      <width>61</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>jpeg</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="bmp">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>10</y>
      <width>61</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>bmp</string>
    </property>
   </widget>
   <widget class="QPushButton" name="applySettings">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>500</y>
      <width>221</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Apply settings</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>240</y>
      <width>211</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertex</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>370</y>
      <width>211</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Facets</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>100</y>
      <width>211</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Background color</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>170</y>
      <width>211</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Projection type</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>300</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Size</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>330</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Type</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>430</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Width</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QComboBox" name="vColor">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>270</y>
      <width>86</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>black</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>yellow</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>purple</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="fColor">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>400</y>
      <width>86</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>black</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>yellow</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>purple</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="bColor">
    <property name="geometry">
     <rect>
      <x>1070</x>
      <y>130</y>
      <width>86</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>white</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>black</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>green</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="pType">
    <property name="geometry">
     <rect>
      <x>1070</x>
      <y>200</y>
      <width>86</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>parallel</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>central</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="vType">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>330</y>
      <width>86</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>square</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>circle</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>none</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="fType">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>460</y>
      <width>86</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>solid</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>dashed</string>
     </property>
    </item>
   </widget>
   <widget class="QSpinBox" name="vSize">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>300</y>
      <width>86</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>10</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="fWidth">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>430</y>
      <width>86</width>
      <height>25</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>10</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_18">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>270</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Color</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_19">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>400</y>
      <width>71</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Color</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_17">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>460</y>
      <width>61</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Type</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="defaultSettings">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>530</y>
      <width>221</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Default settings</string>
    </property>
   </widget>
   <widget class="Line" name="line_5">
    <property name="geometry">
     <rect>
      <x>1000</x>
      <y>90</y>
      <width>223</width>
      <height>3</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(0, 0, 0);</string>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="RESET">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>520</y>
      <width>300</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>RESET</string>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>glwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
